soccer/src/soccer/strategy/coach/coach_node.cpp

```diff
@@ -122,8 +134,109 @@ void CoachNode::check_for_play_state_change() {
 
         coach_state_pub_->publish(coach_message);
 
+        publish_static_obstacles();
+
         play_state_has_changed_ = false;
     }
 }
 
+void CoachNode::field_dimensions_callback(const rj_msgs::msg::FieldDimensions::SharedPtr& msg) {
+    current_field_dimensions_ = *msg;
+    have_field_dimensions_ = true;
+    publish_static_obstacles();
+}
+
+void CoachNode::publish_static_obstacles() {
```

Comment:
just for ease of reading, can we split this into two methods, one for the defense box obstacles and another for the physical goal itself? I know this will add lines of code, but it is so much easier to read something like

```
publish_static_obstacles() {
  if (have_field_dims) {
    defense_box_obstacles = create_defense_box_obstacles();
    goal_wall_obstacles = create_goal_wall_obstacles();
    publisher->publish(defense_box_obstacles);
    publisher->publish(goal_wall_obstacles);
  }
}
```
instead of the 100+ lines that this method currently contains